packages/ui/components/form-core/src/FocusMixin.js

```diff
@@ -52,13 +53,43 @@ const FocusMixinImplementation = superclass =>
     connectedCallback() {
       super.connectedCallback();
       this.__registerEventsForFocusMixin();
+      this.__syncAutofocusToFocusableElement();
     }
 
     disconnectedCallback() {
       super.disconnectedCallback();
       this.__teardownEventsForFocusMixin();
     }
 
+    /**
+     * Gets called when an attribute is changed.
+     * @param {String} name
+     * @param {String | null} _old
+     * @param {String | null} value
+     * @protected
+     */
+    attributeChangedCallback(name, _old, value) {
```

Comment:
Is there a particular reason to use attributeChangedCallback? 
Not necessarily wrong ofc, but we usually rely on the `updated` callback from lit for perf/consistency

Reply:
Hi @tlouisse, honestly it's been a while since I actually created the original pull request, so I'm not quite sure why I decided to go with attributeChangedCallback. My suspicion would be, that, as we're dealing with an attribute this felt like the more appropriate place to listen to the change. That being said I've tested the change with updated and it works just as well, so I've added a commit using the updated, this also means the change to FocusMixinTypes is not required anymore.